src/app/components/PasswordForm/index.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  HiddenIcon,
+  VisibleIcon,
+} from "@bitcoin-design/bitcoin-icons-react/outline";
+import TextField from "@components/form/TextField";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+export type Props = {
+  i18nKeyPrefix: string;
+  children?: React.ReactNode;
+  formData: {
+    password: string;
+    passwordConfirmation: string;
+  };
+  setFormData: (formData: {
+    password: string;
+    passwordConfirmation: string;
+  }) => void;
+};
+
+const initialErrors = {
+  password: "",
+  passwordConfirmation: "",
+};
+
+export default function PasswordForm({
+  formData,
+  setFormData,
+  i18nKeyPrefix,
+}: Props) {
+  const [errors, setErrors] = useState(initialErrors);
+  const [passwordView, setPasswordView] = useState(false);
+  const [passwordConfirmationView, setPasswordConfirmationView] =
+    useState(false);
+  const { t } = useTranslation("translation", { keyPrefix: i18nKeyPrefix });
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    if (event.target.name === "password") {
+      setFormData({
+        password: event.target.value.trim(),
+        passwordConfirmation: formData.passwordConfirmation,
+      });
+    }
+    if (event.target.name === "passwordConfirmation") {
+      setFormData({
+        password: formData.password,
+        passwordConfirmation: event.target.value.trim(),
+      });
+    }
+
+    if (event.target.name === "password" && errors.password) {
+      setErrors({ ...errors, password: "" });
+    } else if (
+      event.target.name === "passwordConfirmation" &&
+      errors.passwordConfirmation &&
+      formData.password === event.target.value.trim()
+    ) {
+      setErrors({ ...errors, passwordConfirmation: "" });
+    }
+  }
+
+  function validate() {
+    let password = "";
```

Comment:
can we call this passwordErrorMessage/passwordConfirmationErrorMessage to be a bit more descriptive and indicate that we actually set the error messages here (and not the actual password field values.